File: from-0801-to-0900/0844-backspace-string-compare/solution.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        sr = []
        for c in s:
            if c == '#':
                if sr:
                    sr.pop()
            else:
                sr.append(c)
                
        tr = []
        for c in t:
            if c == '#':
                if tr:
                    tr.pop()
            else:
                tr.append(c)
              
        print(sr, tr)
        return sr == tr
    

File: from-0801-to-0900/0844-backspace-string-compare/solution.go
unc backspaceCompare(s string, t string) bool {
    sr := []byte{}
    for i := 0; i < len(s); i++ {
        if s[i] == '#' {
            if len(sr) > 0 {
                sr = sr[:len(sr)-1]
            }
        } else {
            sr = append(sr, s[i])
        }
    }
    tr := []byte{}
    for i := 0; i < len(t); i++ {
        if t[i] == '#' {
            if len(tr) > 0 {
                tr = tr[:len(tr)-1]
            }
        } else {
            tr = append(tr, t[i])
        }
    }
    if len(tr) != len(sr) {
        return false
    }
    for i := 0; i < len(sr); i++ {
        if sr[i] != tr[i] {
            return false
        }
    }
    return true
}